apps/electron-app/src/renderer/src/components/UpdateNotification.tsx

```
@@ -0,0 +1,207 @@
+      setUpdateAvailable(false);
+      setShowNoUpdate(true);
+      setTimeout(() => setShowNoUpdate(false), 3000);
+    };
+
+    // Add listeners and store cleanup functions
+    const addListener = (channel: string, handler: any) => {
+      window.api.on(channel, handler);
+      listeners.push(() => {
+        // Remove specific listener
+        const api = window.api as any;
+        if (api._events && api._events[channel]) {
```

Biome (1.9.4)

[error] Change to an optional chain.

Unsafe fix: Change to an optional chain.


(lint/complexity/useOptionalChain)